packages/lib/server/getLuckyUser.ts

```diff
@@ -302,7 +559,58 @@ export async function getLuckyUser<
   distributionMethod: DistributionMethod = DistributionMethod.PRIORITIZE_AVAILABILITY,
   { availableUsers, ...getLuckyUserParams }: GetLuckyUserParams<T>
 ) {
-  const { eventType } = getLuckyUserParams;
+  //maybe pass response directly not id
+  const { eventType, routingFormResponseId } = getLuckyUserParams;
+
+  let attributeWeights;
+  let virtualQueuesData;
+
+  if (routingFormResponseId && getLuckyUserParams.eventType.team?.parentId) {
+    const attributeWithEnabledWeights = await prisma.attribute.findFirst({
+      where: {
+        teamId: getLuckyUserParams.eventType.team?.parentId,
+        isWeightsEnabled: true,
+      },
+      select: {
+        id: true,
+        name: true,
+        slug: true,
+        type: true,
```

Comment:
Better to have a wrapper over it that computes virtualQueuesData, and attributesData and pass to getLuckyUser, following the Open/Close principle

Reply:
Agreed, in general question it any time you need to add code to getLuckyUser